go/account_server/main.go

```diff
@@ -86,28 +82,83 @@ func (s *server) GetUserInfo(ctx context.Context, req *accountpb.GetUserInfoRequ
+// xds-enabled gRPC server.
+type grpcServer interface {
+	grpc.ServiceRegistrar
+	reflection.GRPCServer
+	Serve(net.Listener) error
+}
+
 func main() {
-	args := parseArguments()
+	flag.Parse()
 
-	var serverOpts []grpc.ServerOption
-	if args.gcpClientProject != "" {
-		sd := observability.ConfigureStackdriver(args.gcpClientProject)
+	// Parse credentials type from the command line to determine if xDS
+	// credentials are to be used.
+	xdsCreds, err := utility.ParseCredentialsType(*creds)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Using xDS credentials:", xdsCreds)
+
+	// Use insecure credentials by default. But if xDS credentials are specified
+	// on the command line, use xDS credentials with an insecure fallback.
+	serverCreds := insecure.NewCredentials()
+	if xdsCreds {
+		var err error
+		serverCreds, err = xdscreds.NewServerCredentials(xdscreds.ServerOptions{FallbackCreds: insecure.NewCredentials()})
+		if err != nil {
+			log.Fatalf("Failed to create server xDS credentials: %v", err)
+		}
+	}
+	serverOpts := []grpc.ServerOption{grpc.Creds(serverCreds)}
+
+	// Export stats to Stackdriver if a GCP project was specified.
+	if *gcpClientProject != "" {
+		sd := observability.ConfigureStackdriver(*gcpClientProject)
 		defer sd.Flush()
 		defer sd.StopMetricsExporter()
 		serverOpts = append(serverOpts, grpc.StatsHandler(&ocgrpc.ServerHandler{}))
 	}
 
-	lis, err := net.Listen("tcp", ":"+args.port)
+	// Create a listener to serve Account service RPCs on -port.
+	accountListener, err := net.Listen("tcp4", ":"+*port)
 	if err != nil {
-		log.Fatalf("failed to listen: %v", err)
+		log.Fatalf("Failed to listen: %v", err)
 	}
-	s := grpc.NewServer(serverOpts...)
-	accountpb.RegisterAccountServer(s, &server{hostname: utility.GenHostname(args.hostnameSuffix)})
-	reflection.Register(s)
+
+	// Create the Account service implementation.
+	accountServer := &server{hostname: utility.GenHostname(*hostnameSuffix)}
+
+	// Start serving Account service on -port. We use an xDS-enabled gRPC server
+	// when xDS credentials are to be used, and a vanilla gRPC server otherwise.
+	var s grpcServer
+	if xdsCreds {
+		s = xds.NewGRPCServer(serverOpts...)
+	} else {
+		s = grpc.NewServer(serverOpts...)
+	}
+	accountpb.RegisterAccountServer(s, accountServer)
+
+	// Also export the gRPC health check service on the same port as that of the
+	// Wallet service. This will make it easier to configure GCP health checks
+	// with the --use-serving-port flag.
+	//
+	// Note that the GCP health check will not work if an xds-enabled gRPC server
+	// is used in mTLS mode. In this case, the health check on the -admin_port can
+	// be used.
 	healthServer := health.NewServer()
 	healthServer.SetServingStatus("", healthpb.HealthCheckResponse_SERVING)
 	healthpb.RegisterHealthServer(s, healthServer)
-	if err := s.Serve(lis); err != nil {
-		log.Fatalf("failed to serve: %v", err)
+	reflection.Register(s)
```

Comment:
Why does this need reflection, but not in the other servers?

Reply:
In Java, reflection (called `ProtoReflectionService`) is used in all 3 servers

Reply:
I exported reflection on all servers. We need it because it helps with running grpcurl and some examples, I think refer to using grpcurl.